javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/util/ClassLoaderMap.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling.util;
+
+import io.opentelemetry.instrumentation.api.internal.cache.Cache;
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.description.modifier.Ownership;
+import net.bytebuddy.description.modifier.Visibility;
+import net.bytebuddy.dynamic.loading.ClassLoadingStrategy;
+
+class ClassLoaderMap {
+  private static final Cache<ClassLoader, WeakReference<Map<Object, Object>>> data = Cache.weak();
+
+  public static Object get(ClassLoader classLoader, Object key) {
+    return getClassLoaderData(classLoader).get(key);
```

Comment:
[nitpick] might consider adding a `boolean initialize` parameter to `getClassLoaderData` so that we don't do the heavy initialization work when just checking for the presence of a key.

If `initialize = false`, `getClassLoaderData` will return `null` if the map for the classloader hasn't been initialized yet.